File: cdap-ui/app/cdap/api/__mocks__/metadata.js
/*
 * Copyright © 2017 Cask Data, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 */

/*
  TODO: This is just a stub(mock) for jest to not invoke the actual socket connection.
  This needs to be exported as a singleton class. Will do when we actually need to mock a function.
*/
import Rx from 'rx';

const MyMetadataApi = {
  __metadata: {},
  __properties: {},
  __tags: []
};


MyMetadataApi.__setMetadata = function(metadata, isError) {
  this.__isError = isError;
  this.__metadata = metadata;
};

MyMetadataApi.__setProperties = function(properties, isError) {
  this.__isError = isError;
  this.__properties = properties;
};
MyMetadataApi.__setTags = function(tags, isError) {
  this.__isError = isError;
  this.__tags = tags;
};


MyMetadataApi.generalGetter = function(property) {
  return function() {
    let subject = new Rx.Subject();
    setTimeout(() => {
      if (this.__isError) {
        subject.onError(this[property]);
        return;
      }
      subject.onNext(this[property]);
    });
    return subject;
  }.bind(this);
};
MyMetadataApi.resetState = function() {
  this.__metadata = {};
  this.__properties = {};
  this.__isError = false;
};
MyMetadataApi.getProperties = MyMetadataApi.generalGetter('__properties');
MyMetadataApi.getMetadata = MyMetadataApi.generalGetter('__metadata');
MyMetadataApi.getTags = MyMetadataApi.generalGetter('__tags');

module.exports = {MyMetadataApi};

File: cdap-api/src/main/java/co/cask/cdap/api/metadata/Metadata.java
/*
 * Copyright © 2016 Cask Data, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 */

package co.cask.cdap.api.metadata;

import co.cask.cdap.api.annotation.Beta;

import java.util.Map;
import java.util.Objects;
import java.util.Set;

/**
 * Represents metadata (properties and tags) of an entity.
 */
@Beta
public class Metadata {
  private final Map<String, String> properties;
  private final Set<String> tags;

  public Metadata(Map<String, String> properties, Set<String> tags) {
    this.properties = properties;
    this.tags = tags;
  }

  public Map<String, String> getProperties() {
    return properties;
  }

  public Set<String> getTags() {
    return tags;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (!(o instanceof Metadata)) {
      return false;
    }
    Metadata that = (Metadata) o;
    return Objects.equals(properties, that.properties) &&
      Objects.equals(tags, that.tags);
  }

  @Override
  public int hashCode() {
    return Objects.hash(properties, tags);
  }

  @Override
  public String toString() {
    return "Metadata{" +
      "properties=" + properties +
      ", tags=" + tags +
      '}';
  }
}
